index.bzl

```diff
@@ -53,6 +55,10 @@ js_library = _js_library
 directory_file_path = _directory_file_path
 # ANY RULES ADDED HERE SHOULD BE DOCUMENTED, see index.for_docs.bzl
 
+# Structs for creating custom rules
+nodejs_binary_struct = _nodejs_binary_struct
```

Comment:
I wonder if these should be private APIs to start with (just omit from here and reach into /internal in the impl to start)

also "struct" feels like a strange name to me since it describes the datatype rather than the content. Maybe nodejs_binary_internals or nodejs_binary_impl ? what's the naming convention in that ruleset we're borrowing the concept from?

Reply:
Updated from *_struct to *_kwargs as is used in rules_go: https://github.com/bazelbuild/rules_go/blob/4cd45a2ac59bd00ba54d23ebbdb7e5e2aed69007/go/private/rules/test.bzl#L179